storage/client/queries.go

```diff
@@ -44,17 +44,30 @@ func (qf QueryFactory) BlockQuery() string {
 
 func (qf QueryFactory) TransactionsQuery() string {
 	return fmt.Sprintf(`
-		SELECT block, tx_hash, sender, nonce, fee_amount, method, body, code
-			FROM %s.transactions
-			WHERE ($1::bigint IS NULL OR block = $1::bigint) AND
-						($2::text IS NULL OR method = $2::text) AND
-						($3::text IS NULL OR sender = $3::text) AND
-						($4::bigint IS NULL OR fee_amount >= $4::bigint) AND
-						($5::bigint IS NULL OR fee_amount <= $5::bigint) AND
-						($6::bigint IS NULL OR code = $6::bigint)
-		ORDER BY block DESC, tx_index
-		LIMIT $7::bigint
-		OFFSET $8::bigint`, qf.chainID)
+		SELECT 
+				%[1]s.transactions.block as block,
+				%[1]s.transactions.tx_hash as tx_hash,
+				%[1]s.transactions.sender as sender,
+				%[1]s.transactions.nonce as nonce,
+				%[1]s.transactions.fee_amount as fee_amount,
+				%[1]s.transactions.method as method,
+				%[1]s.transactions.body as body,
+				%[1]s.transactions.code as code
+			FROM %[1]s.transactions
+			LEFT JOIN %[1]s.accounts_related_transactions ON %[1]s.transactions.block = %[1]s.accounts_related_transactions.tx_block 
+				AND %[1]s.transactions.tx_index = %[1]s.accounts_related_transactions.tx_index
+				AND $4::text IS NOT NULL
+				AND %[1]s.accounts_related_transactions.account_address = $4::text
```

Comment:
Do we need these two? They don't really have to do with the join. We filter on the related account in the `WHERE` clause already: `($4::text IS NULL OR %[1]s.accounts_related_transactions.account_address = $4::text)`

Reply:
Technically we don't for the query to work. I included these two in an attempt to make the query more efficient; my thinking was that that if the `ON` clause fails (ie if $4::text is null) then postgres would optimize and skip the join. I need to do more reading/testing to see if that's true though.

Reply:
Uhh actually there's potential for a subtle bug here. Let's leave the `$4::text IS NOT NULL` and a comment:
```
-- When related_address ($4) is NULL and hence we do no filtering on it, avoid the join altogether.
-- Otherwise, every tx will be returned as many times as there are related addresses for it. 
AND $4::text IS NOT NULL
```

Reply:
Sounds good, updated!